core/aws-core/src/main/java/software/amazon/awssdk/awscore/endpoint/AwsClientEndpointProvider.java

```diff
@@ -129,6 +130,7 @@ private Optional<ClientEndpoint> clientEndpointFromEnvironment(Builder builder)
                                           () -> systemProperty(GLOBAL_ENDPOINT_OVERRIDE_SYSTEM_PROPERTY),
                                           () -> environmentVariable(builder.serviceEndpointOverrideEnvironmentVariable),
                                           () -> environmentVariable(GLOBAL_ENDPOINT_OVERRIDE_ENVIRONMENT_VARIABLE),
+                                          () -> servicesProperty(builder),
                                           () -> profileProperty(builder,
                                                                 builder.serviceProfileProperty + "."
                                                                 + ProfileProperty.ENDPOINT_URL),
```

Comment:
Let's add a comment here for context about why we support

```
s3 =
    endpoint_url = ...
```

in both `profile` and `services`